addons/test/misra-test.c

```diff
@@ -273,6 +273,24 @@ void misra_13_6() {
 
 void misra_14_1() {
   for (float f=0.1f; f<1.0f; f += 0.1f){} // 14.1
+  f = 0.0f;
```

Comment:
this `f` seems to be out of scope. maybe `f` should be a local variable in this function instead so it can be used in the while loop conditions.